pants_jupyter_plugin/pex.py

```diff
@@ -185,8 +171,6 @@ def run_pex_tool(args: Iterable[str], **subprocess_args: Any) -> bytes:
             stdout=subprocess.PIPE,
             check=True,
         )
-        for path in result.stdout.decode().splitlines():
-            path_entry = Path(path)
-            sys.path.append(path)
-            self.mounted.append(path_entry)
+
+        for path_entry in self._env_mgr.mount(Path(p) for p in result.stdout.decode().splitlines()):
```

Comment:
FWIW modern Pex supports `from __pex__ import anything.in.a.pex.that.is.on.sys.path`. No un-mounting mechanism is exposed, but its there in the machinery that supports the importing.